### ISSUE TITLE
Difference in model feature importance and shap summary plot

### ISSUE THREAD

**Issue by PranayMehta** [shap/shap#2387]
I have been playing around the toy dataset to understand more about `shap` library and usage. I found this issue that the feature importances from the catboost regressor model is different than the features importances from the `summary_plot` in the `shap` library.

I am analyzing the feature importance from the `model.feature_importances_` on X_train set and the summary plot from `shap explainer` on X_test set.

Here is my source code -
```
import catboost
from catboost import *
import shap
shap.initjs()
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split

X,y = shap.datasets.boston()
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)

# Train Model
model = CatBoostRegressor(iterations=300, learning_rate=0.1, random_seed=123)
model.fit(X_train, y_train, verbose=False, plot=False)

# Compute feature importance dataframe
feat_imp_list = list(zip ( list(model.feature_importances_) , model.feature_names_) )
feature_imp_df = pd.DataFrame(sorted(feat_imp_list, key=lambda x: x[0], reverse=True) , columns = ['feature_value','feature_name'])
feature_imp_df
```
<img width="266" alt="image" src="https://user-images.githubusercontent.com/23047253/153694907-864d9918-6745-4843-84c1-72bf53ed42ac.png">

# Plot feature importance data from model on a bar chart
<img width="617" alt="image" src="https://user-images.githubusercontent.com/23047253/153694912-7caecbcf-4edc-4c94-af26-fc7c823d2b5a.png">

```
# Run shap explainer on X_test set
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X_test)
```

# Plot shap summary plot on X_test
```shap.summary_plot(shap_values, X_test, max_display = X_test.shape[1])```
<img width="700" alt="image" src="https://user-images.githubusercontent.com/23047253/153694921-64f4a0b3-4a94-49eb-9887-93ff01dbc484.png">

# Observations
Why does DIS show up at rank 3 in the feature importance plot from Model but shows up at rank 7 in the summary plot from the SHAP library?

**Comment by leliopusceddu** [comment]
In the article for treeSHAP it is explained how many ways to evaluate the importance of features by weight or gain (for example) fail the properties of consistency and accuracy, only SHAP respects these two properties. In particular, SHAP always calculates the importance taking into account all the possible combinations, in each of which it distributes the score equally given the prediction being studied with respect to the base value.
Another example, I tried to evaluate in the case of the KernelSHAP with a FastSSVM, from the regression to the ranking, a sort of proportionality and consistency between the feature importance for the model and the calculated SHAP values: what you notice is the feature match from regression to ranking it decreases, in this case I think it is due to the assumption of feature independence for the kernelSHAP. From the regression to the ranking the range of the weight distribution becomes wider and wider, (compared to the regression in which the weights are more regularized around zero), this probably complicates the explanation given the independence of the features.

**Comment by rebeen** [comment]
I have encountered the issue of obtaining different rankings for important features when using different SHAP explainers, such as the TreeExplainer and the KernelExplainer.
